File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.Arrays;
import java.util.Set;
import java.util.HashSet;
import java.util.stream.Collectors;

class Solution {
     public int solution(int[] nums) {
        //nums 길이의 절반
        int halfLen = nums.length/2;
        Set numSet = Arrays.stream(nums).boxed().collect(Collectors.toSet());

        int setSize = numSet.size();
         
        return Math.min(halfLen,setSize);

    }
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.cpp
#include <vector>
#include <set>

using namespace std;

int solution(vector<int> nums)
{
    int target = nums.size()/2;
    set<int> max;
    for (int n : nums) {
        max.insert(n);
    }
    int min = max.size();
    if (target < min) {
        return target;
    }
    return min;
}